Sources/BraintreeCore/Configuration/ConfigurationLoader.swift

```diff
@@ -35,56 +40,38 @@ class ConfigurationLoader {
     ///   - `BTConfiguration?`: The configuration object if it is successfully fetched or retrieved from the cache.
     ///   - `Error?`: An error object if fetching the configuration fails or if the instance is deallocated.
     @_documentation(visibility: private)
-    func getConfig(completion: @escaping (BTConfiguration?, Error?) -> Void) {
+    @ConfigurationActor
+    func getConfig() async throws -> BTConfiguration {
+        if let existingTask {
```

Comment:
Shouldn't we check the cache before bothering checking if there are any pending tasks?

@jaxdesmarais @KunJeongPark

Reply:
Jax had the same question. I was thinking of scenario where there may be read/write cache race condition if there is a pending task.